.github/workflows/lint.yml

```diff
@@ -122,17 +124,30 @@ jobs:
       - name: Run flake8
         run: |
           set -eux
+          pip install typing-extensions # for tools/translate_annotations.py
           pip install -r requirements-flake8.txt
           flake8 --version
-          flake8 | tee "${GITHUB_WORKSPACE}"/flake8-output/warnings.txt
+          flake8 | tee "${GITHUB_WORKSPACE}"/flake8-output.txt
+          cp flake8-output.txt flake8-output/annotations.json
+      - name: Translate annotations
+        if: github.event_name == 'pull_request'
+        run: |
+          tools/translate_annotations.py \
+            --file=flake8-output.txt \
+            --regex='^(?P<filename>.*?):(?P<lineNumber>\d+):(?P<columnNumber>\d+): (?P<errorCode>\w+\d+) (?P<errorDesc>.*)' \
```

Comment:
curious--is there a distinction between using single quotes for this line and double quotes for the next? (versus double for both)

Reply:
[This SO answer](https://stackoverflow.com/a/6697781/5044950) explains it pretty well; basically, double quotes expand some things inside them, so we need them for the next line because it uses the value of the `HEAD_SHA` variable; but for this line with the regex, I want to be sure that the value is literally what I've written here, so I use single quotes to prevent any funny business.